### DISCUSSION TITLE
void ImGui::ErrorCheckNewFrameSanityChecks(): Assertion `g.IO.DisplaySize.x >= 0.0f && g.IO.DisplaySize.y >= 0.0f && "Invalid DisplaySize value!"' failed.

### DISCUSSION THREAD

**Discussion by JulioJPinto** [ocornut/imgui#7625] [New Users Build/Link/Run issues ONLY!]
Im currently trying to implement ImGUI in an Engine im developing. Unfortunately everytime i try to run it its not working and I cant figure out why.

Im using OpenGL and Freeglut. Im following the example for opengl_2 and glut but it simply isnt working when i run it with my engine.

here are my code functions for setup, render and shutdown
```
void setupMenu()
{
    // Setup Dear ImGui context
    IMGUI_CHECKVERSION();
    ImGui::CreateContext();

    // Setup Dear ImGui style
    ImGui::StyleColorsDark();

    // Setup Platform/Renderer backends
    ImGui_ImplGLUT_Init();
    ImGui_ImplOpenGL3_Init();
}

void shutDownMenu()
{
    
    ImGui_ImplOpenGL3_Shutdown();
    ImGui_ImplGLUT_Shutdown();
    ImGui::DestroyContext();
}


void renderMenu() {
    ImGui_ImplOpenGL3_NewFrame();
    ImGui_ImplGLUT_NewFrame();
    ImGui::NewFrame();

    ImGuiIO& io = ImGui::GetIO();

    ImGui::Begin("General Controls");

    ImGui::Text("FPS: %f", io.Framerate);

    ImGui::End();

    ImGui::Render();
    io = ImGui::GetIO();
    glViewport(0, 0, (GLsizei) io.DisplaySize.x, (GLsizei) io.DisplaySize.y);
    ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
}
``` 
Im rendering the menu after my models and i want it to be reshapable like the one in the example

**Comment by PathogenDavid** [comment]
It doesn't look like you're installing the GLUT callbacks, see the comments here:

https://github.com/ocornut/imgui/blob/b9084949bd70dcc5ec5f3f417e66db613746f1c7/backends/imgui_impl_glut.h#L34-L44

But more importantly:

https://github.com/ocornut/imgui/blob/b9084949bd70dcc5ec5f3f417e66db613746f1c7/backends/imgui_impl_glut.h#L4-L6

Please use literally anything other than GLUT.

(To FreeGLUT's credit, it did get revived after it got lumped into that statement. However it's still the least featured Dear ImGui backend and very few people use it.)

**Reply by JulioJPinto** [reply]
Thanks for the help! I will look into it. Unfortunately this is for a College project where we were taught to use GLUT and it's due rather soon, so changing for something like GLFW isn't an option at this point. If i manage to fix it i will let it know here :+1:

**Reply by JulioJPinto** [reply]
Thank you so much for the help. We ended up managing to do some simple menus to give information about our scenes and to do some toggles. 
![imgui](https://github.com/ocornut/imgui/assets/94169007/09034685-699f-4332-af28-18540ecedd3e)

**Reply by PathogenDavid** [reply]
Glad I could help! Looks like a cool project, best of luck with your degree!
